node/resolvers/Mutations/Users.ts

```diff
@@ -509,6 +509,44 @@ const Users = {
       })
   },
 
+  createUser: async (
+    _: void,
+    { orgId, costId, roleId, name, email, canImpersonate }: UserArgs,
+    ctx: Context
+  ) => {
+    const {
+      clients: { storefrontPermissions: storefrontPermissionsClient },
+      vtex: { logger },
+    } = ctx as any
+
+    return storefrontPermissionsClient
+      .addUser({ orgId, costId, roleId, name, email, canImpersonate })
+      .then((result: any) => {
+        return result.data.addUser
+      })
+      .catch((error: any) => {
+        logger.error({
+          error,
+          message: 'addUser-error',
+        })
+
+        const message = error.graphQLErrors[0]?.message ?? error.message
+        let status = ''
+
+        if (message.includes(MessageSFPUserAddError.DUPLICATED)) {
+          status = StatusAddUserError.DUPLICATED
+        } else if (
+          message.includes(MessageSFPUserAddError.DUPLICATED_ORGANIZATION)
+        ) {
+          status = StatusAddUserError.DUPLICATED_ORGANIZATION
+        } else {
+          status = StatusAddUserError.ERROR
+        }
+
+        return { status, message }
+      })
```

Comment:
I know that this repo sometimes uses promises and sometimes uses async/await, but in 99.99% of cases we avoid the Promise syntax (async/await is more readable and so it's the usual pattern in VTEX and outside it as well). Could you switch to async/await syntax? It's always possible, since it's just a syntax thing. It would be something like this:

```suggestion
    try {
      const result = await storefrontPermissionsClient
        .addUser({ orgId, costId, roleId, name, email, canImpersonate })
      return result.data.addUser
    } catch (error: any) {
      logger.error({
        error,
        message: 'addUser-error',
      })

      const message = error.graphQLErrors[0]?.message ?? error.message
      let status = ''

      if (message.includes(MessageSFPUserAddError.DUPLICATED)) {
        status = StatusAddUserError.DUPLICATED
      } else if (
        message.includes(MessageSFPUserAddError.DUPLICATED_ORGANIZATION)
      ) {
        status = StatusAddUserError.DUPLICATED_ORGANIZATION
      } else {
        status = StatusAddUserError.ERROR
      }

      return { status, message }
    }
```

Reply:
Sounds good! I just copied the code from `addUser`, but in this case I see no harm in changing it, as one of the ideas is to have a "revamped"/better `addUser` function